src/udev/udevd.c

```diff
@@ -1487,23 +1487,22 @@ static int synthesize_change(sd_device *dev) {
 }
 
 static int on_inotify(sd_event_source *s, int fd, uint32_t revents, void *userdata) {
-        Manager *manager = userdata;
+        Manager *manager = ASSERT_PTR(userdata);
         union inotify_event_buffer buffer;
         ssize_t l;
         int r;
 
-        assert(manager);
-
         r = event_source_disable(manager->kill_workers_event);
         if (r < 0)
                 log_warning_errno(r, "Failed to disable event source for cleaning up idle workers, ignoring: %m");
 
         l = read(fd, &buffer, sizeof(buffer));
         if (l < 0) {
                 if (ERRNO_IS_TRANSIENT(errno))
-                        return 1;
+                        return 0;
 
-                return log_error_errno(errno, "Failed to read inotify fd: %m");
+                log_warning_errno(errno, "Failed to read inotify fd: %m");
+                return 0;
```

Comment:
hmm, why eat up this error? if we can't read from inotify fd we shouldn't ignore this, otherwise we'll busy-loop?

under which conditions would this return an error which is OK to ignore?

Reply:
Hm, indeed. `ERRNO_IS_TRANSIENT()` is already filtered out. Will drop the change.